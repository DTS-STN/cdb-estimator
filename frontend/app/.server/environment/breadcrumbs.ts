import * as v from 'valibot';

// Default breadcrumbs object
export const defaults = {
  BREADCRUMBS: JSON.stringify([
    {
      label: { en: 'Canada.ca', fr: 'Canada.ca' },
      url: { en: 'https://www.canada.ca/en.html', fr: 'https://www.canada.ca/fr.html' },
    },
    {
      label: { en: 'Benefits', fr: 'Prestations' },
      url: { en: 'https://www.canada.ca/en/services/benefits.html', fr: 'https://www.canada.ca/fr/services/prestations.html' },
    },
    {
      label: { en: 'Disability benefits', fr: "Prestations d'invalidité" },
      url: {
        en: 'https://www.canada.ca/en/services/benefits/disability.html',
        fr: 'https://www.canada.ca/fr/services/prestations/handicap.html',
      },
    },
    {
      label: { en: 'Canada Disability Benefit', fr: 'Prestation canadienne pour les personnes handicapées' },
      url: {
        en: 'https://www.canada.ca/en/services/benefits/disability/canada-disability-benefit.html',
        fr: 'https://www.canada.ca/fr/services/prestations/handicap/prestation-canadienne-personnes-situation-handicap.html',
      },
    },
  ] as BreadcrumbItem[]),
} as const;

export interface BreadcrumbItem {
  label: {
    en: string;
    fr: string;
  };
  url: {
    en: string;
    fr: string;
  };
}

// Define schema for a single breadcrumb item
const BreadcrumbSchema = v.object({
  label: v.object({
    en: v.string(),
    fr: v.string(),
  }),
  url: v.object({
    en: v.string(),
    fr: v.string(),
  }),
});

// Define schema for the environment variable
export const breadcrumbs = v.object({
  BREADCRUMBS: v.optional(stringToBreadcrumbsSchema(), defaults.BREADCRUMBS),
});

export type Breadcrumbs = Readonly<v.InferOutput<typeof breadcrumbs>>;

export function stringToBreadcrumbsSchema(): v.GenericSchema<string, BreadcrumbItem[]> {
  return v.pipe(
    v.string(),
    v.transform((raw: string) => {
      try {
        return JSON.parse(raw);
      } catch {
        throw new Error('Invalid JSON in BREADCRUMBS');
      }
    }),
    v.array(BreadcrumbSchema), // Ensures it's an array of valid breadcrumb objects
  );
}
