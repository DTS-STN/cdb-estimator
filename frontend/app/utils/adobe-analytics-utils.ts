type AdobeDataLayer = { push?: (object: Record<string, string | Record<string, string>>) => void };

declare global {
  interface Window {
    adobeDataLayer?: AdobeDataLayer;
  }
}

/**
 *
 * @param value colon seperated identifier value example: 'Dashboard:NextButton'
 * @returns undefined if AA is not enabled. a string otherwise
 */
export function getCustomClick(value: string) {
  if (!isEnabled()) return undefined;

  const { ADOBE_ANALYTICS_CUSTOM_CLICK_PREFIX } = globalThis.__appEnvironment;
  return `${ADOBE_ANALYTICS_CUSTOM_CLICK_PREFIX}:${value}`;
}

export function isEnabled() {
  const { ADOBE_ANALYTICS_ENABLED } = globalThis.__appEnvironment;
  return ADOBE_ANALYTICS_ENABLED;
}

export function isDebug() {
  const { ADOBE_ANALYTICS_DEBUG } = globalThis.__appEnvironment;
  return ADOBE_ANALYTICS_DEBUG;
}

export const getRouteParamsString = (query: Record<string, string | string[] | undefined>) => {
  return Object.entries(query)
    .map(([key, value]) => {
      const sanitizedValue = Array.isArray(value)
        ? value.map((v) => v.replace(/-/g, '_')).join(',') // Handle arrays
        : value?.replace(/-/g, '_'); // Handle single values

      return `${key}-${sanitizedValue}`;
    })
    .join('|');
};

export function pushErrorEvent(errorStatusCode: 404 | 403 | 500) {
  if (!window.adobeDataLayer) {
    console.warn(
      'window.adobeDataLayer is not defined. This could mean your adobe analytics script has not loaded on the page yet.',
    );
    return;
  }

  if (isDebug()) {
    console.debug('Pushing error event to Adobe Data Layer:', errorStatusCode);
  }

  window.adobeDataLayer.push?.({
    event: 'error',
    error: { name: `${errorStatusCode}` },
  });
}

export function pushPageviewEvent(locationUrl: string | URL, query: Record<string, string | string[] | undefined>) {
  if (!window.adobeDataLayer) {
    console.warn(
      'window.adobeDataLayer is not defined. This could mean your adobe analytics script has not loaded on the page yet.',
    );
    return;
  }

  const locationUrlObj = new URL(locationUrl);
  const url = `${locationUrlObj.host}${locationUrlObj.pathname}`; // We don't keep the search params in the URL if any.
  const pageIds = getRouteParamsString(query);

  if (isDebug()) {
    console.debug('Pushing pageLoad event to Adobe Data Layer:', locationUrlObj, url, pageIds);
  }

  window.adobeDataLayer.push?.({
    event: 'pageLoad',
    page: { url, id: pageIds },
  });
}

export function pushValidationErrorEvent(fielIds: readonly string[]) {
  if (!window.adobeDataLayer) {
    console.warn(
      'window.adobeDataLayer is not defined. This could mean your adobe analytics script has not loaded on the page yet.',
    );
    return;
  }

  const joinedFieldIds = fielIds.join('|');
  const { ADOBE_ANALYTICS_ERROR_NAME } = globalThis.__appEnvironment;
  const errorName = `${ADOBE_ANALYTICS_ERROR_NAME}:${joinedFieldIds}`;

  if (isDebug()) {
    console.debug('Pushing error event to Adobe Data Layer:', errorName);
  }

  window.adobeDataLayer.push?.({
    event: 'error',
    error: {
      // Adobe Analytics accepts a maximum of 255 characters
      name: errorName.slice(0, 255),
    },
  });
}
