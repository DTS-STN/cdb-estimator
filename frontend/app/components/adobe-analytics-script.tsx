import * as adobeAnalytics from '~/utils/adobe-analytics-utils';

export function AdobeAnalyticsHeadScript() {
  const { ADOBE_ANALYTICS_JQUERY_SRC, ADOBE_ANALYTICS_SRC } = globalThis.__appEnvironment;

  if (adobeAnalytics.isEnabled()) {
    return (
      <>
        <script src={ADOBE_ANALYTICS_JQUERY_SRC} />
        <script src={ADOBE_ANALYTICS_SRC} />
      </>
    );
  }
  return <></>;
}

export function AdobeAnalyticsBottomScript() {
  if (adobeAnalytics.isEnabled() && adobeAnalytics.isDebug()) {
    return <script id="gc-analytics-bottom-script">_satellite.setDebug(true);</script>;
  }
  return <></>;
}
