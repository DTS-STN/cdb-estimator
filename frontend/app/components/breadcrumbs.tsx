import type { ReactNode } from 'react';

import type { Params, Path, To } from 'react-router';

import { faChevronRight } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import type { ResourceKey, TFunction } from 'i18next';
import { useTranslation } from 'react-i18next';

import { InlineLink } from './links';

import { useLanguage } from '~/hooks/use-language';
import type { I18nRouteFile } from '~/i18n-routes';
import * as adobeAnalytics from '~/utils/adobe-analytics-utils';

export interface BreadcrumbRouteLink {
  file: I18nRouteFile;
  hash?: Path['hash'];
  lang?: Language;
  params?: Params;
  search?: Path['search'];
}

export interface BreadcrumbItem {
  labelKey?: ResourceKey;
  label?: { fr: string; en: string };
  destinationRoute?: BreadcrumbRouteLink;
  destination?: { en: To; fr: To };
}

export interface BreadcrumbsProps {
  className?: string;
  items: BreadcrumbItem[];
}

function getLabel(item: BreadcrumbItem, t: TFunction<['common'], undefined>, currentLanguage?: Language) {
  if (item.labelKey !== undefined) {
    return t(item.labelKey);
  }
  return currentLanguage === 'fr' ? item.label?.fr : item.label?.en;
}

function getDestination(item: BreadcrumbItem, currentLanguage?: Language) {
  if (item.destinationRoute !== undefined) {
    return item.destinationRoute;
  }
  return currentLanguage === 'fr' ? item.destination?.fr : item.destination?.en;
}

export function Breadcrumbs({ className, items }: BreadcrumbsProps) {
  const { t } = useTranslation(['common']);
  const { currentLanguage } = useLanguage();

  return (
    <nav id="wb-bc" className={className} property="breadcrumb" aria-labelledby="breadcrumbs">
      <h2 id="breadcrumbs" className="sr-only">
        {t('common:breadcrumbs.you-are-here')}
      </h2>
      <div className="container">
        <ol className="flex flex-wrap items-center gap-x-3 gap-y-1" typeof="BreadcrumbList">
          {items.map((item, idx) => {
            const label = getLabel(item, t, currentLanguage);
            const destination = getDestination(item, currentLanguage);
            return (
              <li key={label} property="itemListElement" typeof="ListItem" className="flex items-center">
                {idx !== 0 && <FontAwesomeIcon icon={faChevronRight} className="mr-2 size-3 text-slate-700" />}
                <Breadcrumb
                  destination={destination}
                  data-gc-analytics-customclick={adobeAnalytics.getCustomClick(`Breadcrumb:${idx} link`)}
                >
                  {label}
                </Breadcrumb>
              </li>
            );
          })}
        </ol>
      </div>
    </nav>
  );
}

function isBreadcrumbRouteLink(destination: To | BreadcrumbRouteLink): destination is BreadcrumbRouteLink {
  return typeof destination === 'object' && 'file' in destination;
}

function Breadcrumb({ children, destination, ...rest }: { children: ReactNode; destination?: To | BreadcrumbRouteLink }) {
  if (!destination) {
    return <span property="name">{children}</span>;
  }

  return isBreadcrumbRouteLink(destination) ? (
    <InlineLink
      file={destination.file}
      hash={destination.hash}
      params={destination.params}
      search={destination.search}
      property="item"
      typeof="WebPage"
      {...rest}
    >
      <span property="name">{children}</span>
    </InlineLink>
  ) : (
    <InlineLink to={destination} property="item" typeof="WebPage" {...rest}>
      <span property="name">{children}</span>
    </InlineLink>
  );
}
