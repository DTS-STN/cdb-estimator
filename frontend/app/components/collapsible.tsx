import { useId } from 'react';
import type { ComponentProps, ReactNode } from 'react';

import { cn } from '~/utils/tailwind-utils';

export type CollapsibleSummaryProps = ComponentProps<'summary'>;

export function CollapsibleSummary({ children, className, ...props }: CollapsibleSummaryProps) {
  return (
    <summary className={cn('marker:cursor-pointer hover:text-blue-800', className)} {...props}>
      <div className="ml-2 inline-block text-lg font-medium text-slate-950 underline hover:cursor-pointer">{children}</div>
    </summary>
  );
}

export interface CollapsibleProps extends ComponentProps<'details'> {
  contentClassName?: string;
  summary: ReactNode;
}

export function Collapsible({ children, contentClassName, id, summary, ...props }: CollapsibleProps) {
  const uniqueId = useId();
  const summaryId = `${id ?? uniqueId}-summary`;
  const contentId = `${id ?? uniqueId}-content`;
  return (
    <details id={id ?? uniqueId} {...props} className="my-2">
      <CollapsibleSummary id={summaryId}>{summary}</CollapsibleSummary>
      <div id={contentId} className={cn('my-2 border-l-[2px] border-slate-950 px-6 text-lg', contentClassName)}>
        {children}
      </div>
    </details>
  );
}
