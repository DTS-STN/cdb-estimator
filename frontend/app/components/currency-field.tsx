import { useTranslation } from 'react-i18next';
import { NumericFormat } from 'react-number-format';

import type { InputFieldProps } from './input-field';
import { InputField } from './input-field';

import { removeNumericFormatting } from '~/utils/string-utils';
import { cn } from '~/utils/tailwind-utils';

export interface CurrencyFieldProps extends Omit<InputFieldProps, 'type' | 'value' | 'defaultValue'> {
  allowNegative?: boolean;
  value?: string | number | null;
  defaultValue?: string | number | null;
}

export function CurrencyField({ allowNegative = false, maxLength, defaultValue, ...rest }: CurrencyFieldProps) {
  const { i18n } = useTranslation(['common']);
  const strippedDefaultValue = Number(removeNumericFormatting(defaultValue?.toString()));
  const val = !Number.isInteger(strippedDefaultValue) ? strippedDefaultValue.toFixed(2) : strippedDefaultValue;

  return (
    <NumericFormat
      customInput={InputField}
      thousandSeparator={i18n.language === 'fr' ? ' ' : ','}
      allowedDecimalSeparators={i18n.language === 'fr' ? [',', '.'] : ['.']}
      allowNegative={allowNegative}
      decimalScale={2}
      decimalSeparator={i18n.language === 'fr' ? ',' : '.'}
      fixedDecimalScale={false}
      thousandsGroupStyle={'thousand'}
      maxLength={maxLength ?? (i18n.language === 'en' ? 15 : 16)}
      type="text"
      beforeInput={i18n.language === 'en' ? <span className="mr-1">$</span> : undefined}
      afterInput={i18n.language === 'fr' ? <span className="ml-1">$</span> : undefined}
      className={cn('inline', rest.className)}
      defaultValue={val}
      {...rest}
    />
  );
}
