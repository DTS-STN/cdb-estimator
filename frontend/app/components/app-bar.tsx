import { Menu } from '~/components/menu';

interface AppBarProps {
  children: React.ReactNode;
}

export function AppBar({ children }: AppBarProps) {
  return (
    <div className="bg-slate-700">
      <div className="align-center container mx-auto flex flex-wrap justify-between">
        <div className="align-center flex">
          <Menu>{children}</Menu>
        </div>
      </div>
    </div>
  );
}
