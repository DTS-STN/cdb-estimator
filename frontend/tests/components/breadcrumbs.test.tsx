import { createRoutesStub } from 'react-router';

import { render } from '@testing-library/react';
import { useTranslation } from 'react-i18next';
import { afterEach, describe, expect, it, vi } from 'vitest';

import type { BreadcrumbItem } from '~/components/breadcrumbs';
import { Breadcrumbs } from '~/components/breadcrumbs';
import * as adobeAnalytics from '~/utils/adobe-analytics-utils';

vi.mock('~/utils/adobe-analytics-utils', () => ({
  __esModule: true,
  isEnabled: vi.fn(),
  getCustomClick: vi.fn(),
}));

describe('Breadcrumbs', () => {
  afterEach(() => {
    vi.mocked(useTranslation).mockReset();
    vi.mocked(adobeAnalytics.isEnabled).mockReset();
    vi.mocked(adobeAnalytics.getCustomClick).mockReset();
  });

  it.each(
    new Array<{ lang: Language; items: BreadcrumbItem[] }>(
      {
        lang: 'en',
        items: [] as BreadcrumbItem[],
      },
      {
        lang: 'fr',
        items: [
          {
            label: { en: 'Canada.ca en', fr: 'Canada.ca fr' },
            destination: { en: 'https://www.canada.ca/en.html', fr: 'https://www.canada.ca/fr.html' },
          },
        ] as BreadcrumbItem[],
      },
      {
        lang: 'en',
        items: [
          {
            label: { en: 'Canada.ca en', fr: 'Canada.ca fr' },
            destination: { en: 'https://www.canada.ca/en.html', fr: 'https://www.canada.ca/fr.html' },
          },
        ] as BreadcrumbItem[],
      },
      {
        lang: 'fr',
        items: [{ labelKey: 'estimator:index.breadcrumb', destinationRoute: { file: 'routes/index.tsx' } }] as BreadcrumbItem[],
      },
      {
        lang: 'en',
        items: [{ labelKey: 'estimator:index.breadcrumb', destinationRoute: { file: 'routes/index.tsx' } }] as BreadcrumbItem[],
      },
      {
        lang: 'en',
        items: [
          {
            label: { en: 'Canada.ca', fr: 'Canada.ca' },
            destination: { en: 'https://www.canada.ca/en.html', fr: 'https://www.canada.ca/fr.html' },
          },
          {
            label: { en: 'Benefits', fr: 'Prestations' },
            destination: {
              en: 'https://www.canada.ca/en/services/benefits.html',
              fr: 'https://www.canada.ca/fr/services/prestations.html',
            },
          },
          {
            label: { en: 'Disability benefits', fr: "Prestations d'invalidité" },
            destination: {
              en: 'https://www.canada.ca/en/services/benefits/disability.html',
              fr: 'https://www.canada.ca/fr/services/prestations/handicap.html',
            },
          },
          {
            label: { en: 'Canada Disability Benefit', fr: 'Prestation canadienne pour les personnes handicapées' },
            destination: {
              en: 'https://www.canada.ca/en/services/benefits/disability/canada-disability-benefit.html',
              fr: 'https://www.canada.ca/fr/services/prestations/handicap/prestation-canadienne-personnes-situation-handicap.html',
            },
          },
          { labelKey: 'estimator:index.breadcrumb', destinationRoute: { file: 'routes/index.tsx' } },
        ] as BreadcrumbItem[],
      },
    ),
  )('should render links', ({ lang, items }) => {
    const translation = useTranslation();
    vi.mocked(useTranslation).mockReturnValue({ ...translation, i18n: { ...translation.i18n, language: lang } });
    vi.mocked(adobeAnalytics.isEnabled).mockReturnValue(false);
    const getCustomClickMock = vi.mocked(adobeAnalytics.getCustomClick).mockReturnValue(undefined);

    const RoutesStub = createRoutesStub([
      {
        path: `/${lang}`,
        Component: () => <Breadcrumbs items={items} />,
      },
    ]);

    const { container } = render(<RoutesStub initialEntries={[`/${lang}`]} />);
    expect(container).toMatchSnapshot('expected html');
    expect(getCustomClickMock).toBeCalledTimes(items.length);
  });
});
