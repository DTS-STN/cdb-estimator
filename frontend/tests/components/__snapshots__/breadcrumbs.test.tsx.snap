// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Breadcrumbs > should render links > expected html 1`] = `
<div>
  <nav
    aria-labelledby="breadcrumbs"
    id="wb-bc"
    property="breadcrumb"
  >
    <h2
      class="sr-only"
      id="breadcrumbs"
    >
      common:breadcrumbs.you-are-here
    </h2>
    <div
      class="container"
    >
      <ol
        class="flex flex-wrap items-center gap-x-3 gap-y-1"
        typeof="BreadcrumbList"
      />
    </div>
  </nav>
</div>
`;

exports[`Breadcrumbs > should render links > expected html 2`] = `
<div>
  <nav
    aria-labelledby="breadcrumbs"
    id="wb-bc"
    property="breadcrumb"
  >
    <h2
      class="sr-only"
      id="breadcrumbs"
    >
      common:breadcrumbs.you-are-here
    </h2>
    <div
      class="container"
    >
      <ol
        class="flex flex-wrap items-center gap-x-3 gap-y-1"
        typeof="BreadcrumbList"
      >
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            href="https://www.canada.ca/fr.html"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              Canada.ca fr
            </span>
          </a>
        </li>
      </ol>
    </div>
  </nav>
</div>
`;

exports[`Breadcrumbs > should render links > expected html 3`] = `
<div>
  <nav
    aria-labelledby="breadcrumbs"
    id="wb-bc"
    property="breadcrumb"
  >
    <h2
      class="sr-only"
      id="breadcrumbs"
    >
      common:breadcrumbs.you-are-here
    </h2>
    <div
      class="container"
    >
      <ol
        class="flex flex-wrap items-center gap-x-3 gap-y-1"
        typeof="BreadcrumbList"
      >
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            href="https://www.canada.ca/en.html"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              Canada.ca en
            </span>
          </a>
        </li>
      </ol>
    </div>
  </nav>
</div>
`;

exports[`Breadcrumbs > should render links > expected html 4`] = `
<div>
  <nav
    aria-labelledby="breadcrumbs"
    id="wb-bc"
    property="breadcrumb"
  >
    <h2
      class="sr-only"
      id="breadcrumbs"
    >
      common:breadcrumbs.you-are-here
    </h2>
    <div
      class="container"
    >
      <ol
        class="flex flex-wrap items-center gap-x-3 gap-y-1"
        typeof="BreadcrumbList"
      >
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            data-discover="true"
            href="/fr"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              estimator:index.breadcrumb
            </span>
          </a>
        </li>
      </ol>
    </div>
  </nav>
</div>
`;

exports[`Breadcrumbs > should render links > expected html 5`] = `
<div>
  <nav
    aria-labelledby="breadcrumbs"
    id="wb-bc"
    property="breadcrumb"
  >
    <h2
      class="sr-only"
      id="breadcrumbs"
    >
      common:breadcrumbs.you-are-here
    </h2>
    <div
      class="container"
    >
      <ol
        class="flex flex-wrap items-center gap-x-3 gap-y-1"
        typeof="BreadcrumbList"
      >
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            data-discover="true"
            href="/en"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              estimator:index.breadcrumb
            </span>
          </a>
        </li>
      </ol>
    </div>
  </nav>
</div>
`;

exports[`Breadcrumbs > should render links > expected html 6`] = `
<div>
  <nav
    aria-labelledby="breadcrumbs"
    id="wb-bc"
    property="breadcrumb"
  >
    <h2
      class="sr-only"
      id="breadcrumbs"
    >
      common:breadcrumbs.you-are-here
    </h2>
    <div
      class="container"
    >
      <ol
        class="flex flex-wrap items-center gap-x-3 gap-y-1"
        typeof="BreadcrumbList"
      >
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            href="https://www.canada.ca/en.html"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              Canada.ca
            </span>
          </a>
        </li>
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-right mr-2 size-3 text-slate-700"
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
              fill="currentColor"
            />
          </svg>
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            href="https://www.canada.ca/en/services/benefits.html"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              Benefits
            </span>
          </a>
        </li>
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-right mr-2 size-3 text-slate-700"
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
              fill="currentColor"
            />
          </svg>
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            href="https://www.canada.ca/en/services/benefits/disability.html"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              Disability benefits
            </span>
          </a>
        </li>
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-right mr-2 size-3 text-slate-700"
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
              fill="currentColor"
            />
          </svg>
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            href="https://www.canada.ca/en/services/benefits/disability/canada-disability-benefit.html"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              Canada Disability Benefit
            </span>
          </a>
        </li>
        <li
          class="flex items-center"
          property="itemListElement"
          typeof="ListItem"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-right mr-2 size-3 text-slate-700"
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
              fill="currentColor"
            />
          </svg>
          <a
            class="text-slate-700 underline hover:text-blue-700 focus:text-blue-700"
            data-discover="true"
            href="/en"
            property="item"
            typeof="WebPage"
          >
            <span
              property="name"
            >
              estimator:index.breadcrumb
            </span>
          </a>
        </li>
      </ol>
    </div>
  </nav>
</div>
`;
